tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -648,188 +744,214 @@ bool DeclResultIdMapper::decorateResourceBindings() {
   return noError;
 }
 
-QualType
-DeclResultIdMapper::getFnParamOrRetType(const DeclaratorDecl *decl) const {
-  if (const auto *funcDecl = dyn_cast<FunctionDecl>(decl)) {
-    return funcDecl->getReturnType();
-  }
-  return decl->getType();
-}
+bool DeclResultIdMapper::createStageVars(
+    const DeclaratorDecl *decl, const hlsl::SigPoint *sigPoint, bool asInput,
+    QualType type, uint32_t arraySize, llvm::Optional<uint32_t> invocationId,
+    uint32_t *value, const llvm::Twine &namePrefix) {
+  // invocationId should only be used for handling HS per-vertex output.
+  assert(invocationId.hasValue() ? shaderModel.IsHS() : true);
+  assert(invocationId.hasValue() ? !asInput : true);
+  assert(invocationId.hasValue() ? arraySize != 0 : true);
```

Comment:
Was initially written like that to avoid having empty `if` statements for release builds. But I think it does not really matter since the compiler should optimize empty `if`s away. Will change.

Reply:
changed